src/main/java/it/gov/pagopa/afm/utils/ImportCDIFunction.java

```diff
@@ -0,0 +1,167 @@
+package it.gov.pagopa.afm.utils;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+
+import org.apache.commons.lang3.SerializationUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+
+import com.azure.cosmos.implementation.apachecommons.lang.StringUtils;
+
+import feign.FeignException;
+import it.gov.pagopa.afm.utils.entity.CDI;
+import it.gov.pagopa.afm.utils.entity.Detail;
+import it.gov.pagopa.afm.utils.entity.ServiceAmount;
+import it.gov.pagopa.afm.utils.entity.StatusType;
+import it.gov.pagopa.afm.utils.exception.AppError;
+import it.gov.pagopa.afm.utils.exception.AppException;
+import it.gov.pagopa.afm.utils.model.bundle.BundleRequest;
+import it.gov.pagopa.afm.utils.model.bundle.BundleResponse;
+import it.gov.pagopa.afm.utils.model.bundle.BundleType;
+import it.gov.pagopa.afm.utils.model.bundle.Wrapper;
+import it.gov.pagopa.afm.utils.service.CDIService;
+import it.gov.pagopa.afm.utils.service.MarketPlaceClient;
+import lombok.AllArgsConstructor;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import reactor.core.publisher.Mono;
+
+@Component
+@Slf4j
+@NoArgsConstructor
+@AllArgsConstructor
+public class ImportCDIFunction  implements Function<Mono<Wrapper>, Mono<List<BundleResponse>>>{
+
+	@Autowired(required=false)
```

Comment:
why are they false?

Reply:
For a problem with context initialization with JUNIT tests.

To invoke the function from a junit test it's necessary to explicitly instantiate the FunctionInvoker object: `FunctionInvoker<Wrapper object, List<BundleResponse>> handler = new FunctionInvoker<>(ImportCDIFunction.class);` ([microsoft doc here](https://learn.microsoft.com/it-it/azure/developer/java/spring-framework/getting-started-with-spring-cloud-function-in-azure)).
A new object is out of spring context and the only constructor of the FunctionInvoker is with the class reference (work by reflection).
@Autowired to true (default) resulted in a spring exception during object initialization.  I couldn't find a way to avoid the error other than not making autowired mandatory.

In any case the problem is related only to the context of the junit tests, in a real working situation the objects are always available in the context and therefore the autowired works correctly.